Assets/Src/Scripts/Input/WebCamSource.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+using System;
+using System.Diagnostics;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using Debug = UnityEngine.Debug;
+
+namespace SeedAvatar {
+  public enum DropdownKey {
+    CAMERA = 0,
+    RESOLUTION = 1,
+    FPS = 2,
+    FLIP = 3
+  }
+  public enum ResolutionOptions {
+    LOW = 0,
+    MIDDLE = 1,
+    HIGH = 2,
+  }
+  public enum FpsOptions {
+    LOW = 0,
+    HiGH = 1
+  }
+  public enum FlipOptions {
+    NORMAL = 0,
+    MIRRORING = 1
+  }
+  public class WebCamSource : MonoBehaviour {
+    public Dropdown CameraDropdown;
+    public Dropdown ResolutionDropdown;
+    public Dropdown FpsDropdown;
+    public Dropdown FlipDropdown;
+    public string DeviceName;
+    public int Fps = 30;
+    public bool Flip = false;
+    [SerializeField] private RenderTexture _buffer;
+    private WebCamTexture _webCam;
+    private int _width = 1280;
+    private int _height = 720;
+    private static readonly Dictionary<ResolutionOptions, string> _resolutionDic =
+    new Dictionary<ResolutionOptions, string>
+    {
+      { ResolutionOptions.LOW , "1280x720" } , {  ResolutionOptions.MIDDLE, "1600x900" } , { ResolutionOptions.HIGH , "1920x1080" }
+    };
+    private static readonly Dictionary<FpsOptions, string> _fpsDic =
+    new Dictionary<FpsOptions, string>
+    {
+      { FpsOptions.LOW , "30Hz" } , { FpsOptions.HiGH , "60Hz" }
+    };
+    private static readonly Dictionary<FlipOptions, string> _flipDic =
+    new Dictionary<FlipOptions, string>
+    {
+      { FlipOptions.NORMAL , "Normal" } , { FlipOptions.MIRRORING , "Mirroring" }
+    };
+    private List<string> _devicesName = new List<string>();
+    private List<string> _resolutionOpt = new List<string>();
+    private List<string> _fpsOpt = new List<string>();
+    private List<string> _flipOpt = new List<string>();
```

Comment:
69-72: All these four variables are not really class members. See your following code, they are only used for temporary purposes between two method calls. Please do not define this kind of temporary variable in the class scope. We have a lot of much better ways to achieve the same task.